### DISCUSSION TITLE
Call Go/golang library from Micropython?

### DISCUSSION THREAD

**Discussion by mdaeron** [micropython/micropython#10104] [Libraries & Drivers]
I'm aware that it is possible to build and call C-based libraries from Micropython. Is there any way/usefulness to do the same with Go (golang) libraries, perhaps using something like [gopy](https://github.com/go-python/gopy)?

**Comment by jimmo** [comment]
@mdaeron -- This would not be possible on microcontroller targets... Go has a significant runtime implementation, not to mention all the system libraries you would need too.

You could embed TinyGo into MicroPython (or the other way around), but the firmware size would be prohibitive as there would be very little sharing.

**Reply by mdaeron** [reply]
Understood, thanks.

**Comment by vshymanskyy** [comment]
It is possible now: https://github.com/orgs/micropython/discussions/15702
